core/workspace_svg.ts

```diff
@@ -2541,6 +2541,69 @@ export class WorkspaceSvg extends Workspace implements IASTNodeLocationSvg {
       this.removeClass('blocklyReadOnly');
     }
   }
+
+  /**
+   * Scrolls the provided bounds into view.
+   *
+   * In the case of small workspaces/large bounds, this function prioritizes
+   * getting the top left corner of the bounds into view. It also adds some
+   * padding around the bounds to allow the element to be comfortably in view.
+   *
+   * @internal
+   * @param bounds A rectangle to scroll into view, as best as possible.
+   */
+  scrollBoundsIntoView(bounds: Rect) {
+    if (Gesture.inProgress()) {
+      // This can cause jumps during a drag and it only suited for keyboard nav.
+      return;
+    }
+    const scale = this.getScale();
+
+    const rawViewport = this.getMetricsManager().getViewMetrics(true);
+    const viewport = new Rect(
+      rawViewport.top,
+      rawViewport.top + rawViewport.height,
+      rawViewport.left,
+      rawViewport.left + rawViewport.width,
+    );
+
+    if (
+      bounds.left >= viewport.left &&
+      bounds.top >= viewport.top &&
+      bounds.right <= viewport.right &&
+      bounds.bottom <= viewport.bottom
+    ) {
+      // Do nothing if the block is fully inside the viewport.
+      return;
+    }
+
+    // Add some padding to the bounds so the element is scrolled comfortably
+    // into view.
+    bounds = bounds.clone();
+    bounds.top -= 10;
+    bounds.bottom += 10;
+    bounds.left -= 10;
+    bounds.right += 10;
+
+    let deltaX = 0;
+    let deltaY = 0;
+
+    if (bounds.left < viewport.left) {
+      deltaX = viewport.left - bounds.left;
+    } else if (bounds.right > viewport.right) {
+      deltaX = viewport.right - bounds.right;
```

Comment:
If the top left is within the viewport but the bounds are larger than the viewport will this jump to the bottom right of the bounds? I think you need a Math.min(bounds.left - viewport.left, viewport.right - bounds.right) to make sure it doesn't scroll the left boundary outside the viewport.

Same for the top one below.

Reply:
I don't think setting it to zero will work in all cases. If the bounds are bigger than the viewport and are completely to the right of the viewport you wouldn't scroll them into view at all. I think you need the Math.min version to catch all the edge cases.

Reply:
The Math.min() version suffers from the same issue; since this is a bit of an edge case and the method is `@internal`, I filed #8855 to figure out the behavior we want and adjust this if needed. I also reverted the set-to-zero approach.